sparkmagic/sparkmagic/tests/test_sparkstorecommand.py

```diff
@@ -0,0 +1,246 @@
+
+
+@with_setup(_setup, _teardown)
+def test_r_livy_sampling_options():
+    variable_name = "abc"
+
+    sparkcommand = SparkStoreCommand(variable_name, samplemethod='take', maxrows=100)
+
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(take({},100)))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(variable_name, samplemethod='take', maxrows=-1)
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(collect({})))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(variable_name, samplemethod='sample', samplefraction=0.25, maxrows=-1)
+    assert_equals(sparkcommand._r_command(variable_name),
+                  Command('for ({} in (jsonlite::toJSON(collect(sample({}, FALSE, 0.25))))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+    sparkcommand = SparkStoreCommand(variable_name, samplemethod='sample', samplefraction=0.33, maxrows=3234)
+    assert_equals(sparkcommand._r_command(variable_name),
+                 Command('for ({} in (jsonlite::toJSON(take(sample({}, FALSE, 0.33),3234)))) {{cat({})}}'.format(LONG_RANDOM_VARIABLE_NAME, variable_name, LONG_RANDOM_VARIABLE_NAME)))
+
+
+@with_setup(_setup, _teardown)
+def test_execute_code():
+    spark_events = MagicMock()
+    variable_name = "abc"
+
+    sparkcommand = SparkStoreCommand(variable_name, "take", 100, 0.2, spark_events=spark_events)
+    sparkcommand.to_command = MagicMock(return_value=MagicMock())
+    result = """{"z":100, "nullv":null, "y":50}
+{"z":25, "nullv":null, "y":10}"""
+    sparkcommand.to_command.return_value.execute = MagicMock(return_value=(True, result))
+    session = MagicMock()
+    session.kind = "pyspark"
+    result = sparkcommand.execute(session)
+    
+    sparkcommand.to_command.assert_called_once_with(session.kind, variable_name)
+    sparkcommand.to_command.return_value.execute.assert_called_once_with(session)
+
+
+@with_setup(_setup, _teardown)
+def test_unicode():
+    variable_name = u"collect 'è'"
+
+    sparkcommand = SparkStoreCommand(variable_name, samplemethod='take', maxrows=120)
+    assert_equals(sparkcommand._pyspark_command(variable_name),
+                  Command(u'for {} in {}.toJSON().take(120): print({}.encode("{}"))'\
+                          .format(LONG_RANDOM_VARIABLE_NAME, variable_name,
+                                  LONG_RANDOM_VARIABLE_NAME, conf.pyspark_python_encoding())))
+    assert_equals(sparkcommand._scala_command(variable_name),
+                  Command(u'{}.toJSON.take(120).foreach(println)'.format(variable_name)))
+
+
+@with_setup(_setup, _teardown)
+def test_scala_livy_options_spark2():
+    variable_name = "var_name"
```

Comment:
You don't need Spark 2 tests because you are not referencing the name of the sql context, which changes from spark 1 to spark 2.